gstreamer-pbutils/src/auto/discoverer_info.rs

```diff
@@ -0,0 +1,175 @@
+// This file was generated by gir (d933f9a+) from gir-files (???)
+// DO NOT EDIT
+
+use DiscovererResult;
+use DiscovererSerializeFlags;
+use DiscovererStreamInfo;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct DiscovererInfo(Object<ffi::GstDiscovererInfo>);
+
+    match fn {
+        get_type => || ffi::gst_discoverer_info_get_type(),
+    }
+}
+
+impl DiscovererInfo {
+    pub fn from_variant(variant: &glib::Variant) -> Option<DiscovererInfo> {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_discoverer_info_from_variant(variant.to_glib_none().0))
+        }
+    }
+}
+
+unsafe impl Send for DiscovererInfo {}
+unsafe impl Sync for DiscovererInfo {}
+
+pub trait DiscovererInfoExt {
+    fn copy(&self) -> Option<DiscovererInfo>;
+
+    fn get_audio_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_container_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_duration(&self) -> gst::ClockTime;
+
+    fn get_misc(&self) -> Option<gst::Structure>;
+
+    fn get_missing_elements_installer_details(&self) -> Vec<String>;
+
+    fn get_result(&self) -> DiscovererResult;
+
+    fn get_seekable(&self) -> bool;
+
+    fn get_stream_info(&self) -> Option<DiscovererStreamInfo>;
+
+    fn get_stream_list(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_streams(&self, streamtype: glib::types::Type) -> Vec<DiscovererStreamInfo>;
+
+    fn get_subtitle_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_tags(&self) -> Option<gst::TagList>;
+
+    fn get_toc(&self) -> Option<gst::Toc>;
+
+    fn get_uri(&self) -> Option<String>;
```

Comment:
Can this be NULL?

Reply:
I don't think we have checks for NULL URIs on the API. It would eventually be set to NULL and it could end up being reported with an error to the user. Will do a simple test to confirm but this is what I can gather from the code.